File: src/main/java/com/facebook/components/ImageContent.java
/**
 * Copyright (c) 2014-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

package com.facebook.components;

import java.util.Collections;
import java.util.List;

import android.graphics.drawable.Drawable;

/**
 * A UI element that contains simple resource drawables.
 */
public interface ImageContent {

  /**
   * An empty instance of {@link ImageContent}.
   */
  ImageContent EMPTY = new ImageContent() {
    @Override
    public List<Drawable> getImageItems() {
      return Collections.EMPTY_LIST;
    }
  };

  /**
   * @return the list of image drawables that are rendered by this UI element. The list returned
   * should not be modified and may be unmodifiable.
   */
  List<Drawable> getImageItems();
}

File: litho-core/src/main/java/com/facebook/litho/ImageContent.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.facebook.litho

import android.graphics.drawable.Drawable
import kotlin.jvm.JvmField

/** A UI element that contains simple resource drawables. */
interface ImageContent {

  /**
   * @return the list of image drawables that are rendered by this UI element. The list returned
   *   should not be modified and may be unmodifiable.
   */
  val imageItems: List<Drawable>

  companion object {
    /** An empty instance of [ImageContent]. */
    @JvmField
    val EMPTY: ImageContent =
        object : ImageContent {
          override val imageItems: List<Drawable>
            get() = emptyList()
        }
  }
}
